src/handlers/rotate.rs

```diff
@@ -0,0 +1,151 @@
+use crate::util::image::{move_image, remove_cache_entries};
+use crate::{
+    util::{
+        auth::check_api_key,
+        image::{determine_img_dir, determine_img_path, save_image, ImageSearchBehaviour},
+    },
+    ServerState,
+};
+use axum::extract::{Query, State};
+use axum::headers::authorization::Bearer;
+use axum::headers::Authorization;
+use axum::http::StatusCode;
+use axum::TypedHeader;
+use libvips::{ops, VipsImage};
+use serde::Deserialize;
+use std::fs::rename;
+use uuid::Uuid;
+
+#[derive(Deserialize)]
+pub struct RotateQuery {
+    id: Uuid,
+    angle: f64,
+}
+
+pub async fn rotate_handler(
+    State(server_state): State<ServerState>,
+    TypedHeader(authorization): TypedHeader<Authorization<Bearer>>,
+    query: Query<RotateQuery>,
+) -> Result<String, (StatusCode, String)> {
+    check_api_key(authorization, &server_state.api_key_hash)?;
+
+    if query.angle < 0.0 || query.angle > 360.0 {
+        return Err((
+            StatusCode::BAD_REQUEST,
+            "Angle must be 0.0 < angle < 360.0!".to_owned(),
+        ));
+    }
+
+    let image_directory = determine_img_dir(query.id, ImageSearchBehaviour::Valid);
+    if image_directory.is_err() {
+        return Err((StatusCode::NOT_FOUND, "Image not found!".to_owned()));
+    }
+
+    let image_directory = image_directory.unwrap();
+    let image_directory = image_directory.to_str().unwrap();
+
+    let image_path = determine_img_path(image_directory, query.id);
+    if image_path.is_err() {
+        log::warn!(
+            "Image not found where the path was previously determined. Id: {:?}, Directory: {:?}, Error: {:?}",
+            query.id,
+            image_directory,
+            image_path.err()
+        );
+        return Err((StatusCode::NOT_FOUND, "Image not found!".to_owned()));
+    }
+
+    let image_path = image_path.unwrap();
+    let image_path = image_path.to_str().unwrap();
```

Comment:
Same as above, I'm not a huge fan of redefining a variable with the same name multiple times.